plugins/out_loki/loki.c

```diff
@@ -59,28 +59,28 @@ static inline void normalize_cat(struct flb_ra_parser *rp, flb_sds_t name)
     /* Iterate record accessor keys */
     key = rp->key;
     if (rp->type == FLB_RA_PARSER_STRING) {
-        safe_sds_cat(&name, key->name, flb_sds_len(key->name));
+        safe_sds_cat(name, key->name, flb_sds_len(key->name));
     }
     else if (rp->type == FLB_RA_PARSER_KEYMAP) {
-        safe_sds_cat(&name, key->name, flb_sds_len(key->name));
+        safe_sds_cat(name, key->name, flb_sds_len(key->name));
         if (mk_list_size(key->subkeys) > 0) {
-            safe_sds_cat(&name, "_", 1);
+            safe_sds_cat(name, "_", 1);
```

Comment:
There is something about caught my attention, could you clarify this if you have a better understanding?

If the conditional in lines 66-68 append an underscore when there are one or more subkeys but we have a for each block that appends an underscore before each subkey, why do we need to have that previous step and a conditional in each sub key iteration to ensure we don't append a "repeated" underscore before the first one?

Wouldn't it make sense to remove the conditional in lines 66-68 and make the append in line 75 unconditional?
I can't think of a scenario where that initial underscore would be appended but the for each block wouldn't iterate at least once and since it's a prepend and not an append (ie. before the sub key and not after) we wouldn't need to keep track of "is this sub key the last one?".

It's certainly not a change you made in this PR and I'm not holding you accountable for it, I'm just raising the question in case you think it makes sense and we should simplify the code by removing it (since `sub` isn't used for anything else).

Reply:
I don't know why.
If it should be fixed, I prefer to create another PR since I want to fix only pointer handling with this PR.

The commit is 
https://github.com/fluent/fluent-bit/commit/2f0ea62e7ad49fd39d356e03706f2a3bda18a89d